src/task.ts

```diff
@@ -92,37 +119,49 @@ export function task<TResult, TReason = any>(
         });
       });
     },
-  };
-  runner({
-    resolve(result: TResult) {
-      if (!task.isRunning) {
-        // todo: decide whether to swallow or throw
-        return;
+    run(events) {
```

Comment:
Type annotation is missing.

```suggestion
    run(events: TaskEventNotifications<TResult, TReason>) {
```